src/Formats/ProtobufSerializer.cpp

```diff
@@ -3322,125 +3322,139 @@ namespace
                     }
                 }
 
+                /// Complex case: one or more columns are serialized as a nested message.
                 for (const auto & [field_descriptor, suffix] : field_descriptors_with_suffixes)
                 {
-                    if (!suffix.empty())
+                    if (suffix.empty())
+                        continue;
```

Comment:
This specific change is not required per se, but it arguably brings all relevant code paths into the diff.
If it may help the review, we can get rid of that 🙏